src/dmd/expressionsem.d

```diff
@@ -10177,6 +10177,129 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             global.params.useDIP1000 == FeatureState.enabled)
             checkAssignEscape(sc, res, false, false);
         result = res;
+
+        if (!exp.isCTFEHookPart &&
+            (exp.op == EXP.concatenateAssign || exp.op == EXP.concatenateElemAssign) &&
+            !(sc.flags & (SCOPE.ctfe | SCOPE.compile)))
+        {
+            // if aa ordering is triggered, `res` will be a CommaExp
+            // and `.e2` will be the rewritten original expression.
+
+            // `output` will point to the expression that the lowering will overwrite
+            Expression* output;
+            if (auto comma = res.isCommaExp())
+            {
+                output = &comma.e2;
+                // manual cast because it could be either CatAssignExp or CatElemAssignExp
+                exp = cast(CatAssignExp)comma.e2;
+            }
+            else
+            {
+                output = &result;
+                exp = cast(CatAssignExp)result;
+            }
+
+            if (exp.op == EXP.concatenateAssign)
+            {
+                Identifier hook = global.params.tracegc ? Id._d_arrayappendTTrace : Id._d_arrayappendT;
+
+                if (!verifyHookExist(exp.loc, *sc, hook, "appending array to arrays", Id.object))
+                    return setError();
+
+                // Lower to object._d_arrayappendT{,Trace}({file, line, funcname}, e1, e2)
+                 Expression id = new IdentifierExp(exp.loc, Id.empty);
+                id = new DotIdExp(exp.loc, id, Id.object);
+                id = new DotIdExp(exp.loc, id, hook);
+                id = id.expressionSemantic(sc);
```

Comment:
I removed this call and as many calls under `EXP.concatenateElemAssign` as I could. One call is still needed before creating the `CommaExp`. Otherwise, compilation raises a seg fault.